src/managers/cluster.py

```diff
@@ -300,16 +300,17 @@ def move_leader_if_required(self) -> None:
 
     def update_cluster_member_state(self) -> None:
         """Get up-to-date member information and store in cluster state."""
-        client = EtcdClient(
-            username=self.admin_user,
-            password=self.admin_password,
-            client_url=self.state.unit_server.client_url,
-        )
+        if self.state.cluster.cluster_state:
```

Comment:
nit, I think it's better in this case to have the inverted condition and return - to reduce nestedness